File: src/main/java/net/hectus/neobb/turn/person_game/block/PTCake.java
package net.hectus.neobb.turn.person_game.block;

import net.hectus.neobb.buff.Buff;
import net.hectus.neobb.player.NeoPlayer;
import net.hectus.neobb.turn.default_game.block.BlockTurn;
import net.hectus.neobb.turn.person_game.categorization.BuffCategory;
import org.bukkit.block.Block;

import java.util.List;

public class PTCake extends BlockTurn implements BuffCategory {
    public PTCake(NeoPlayer player) { super(player); }
    public PTCake(Block data, NeoPlayer player) { super(data, player); }

    @Override
    public List<Buff> buffs() {
        return List.of(new Buff.Luck(Buff.BuffTarget.YOU, 20));
    }
}

File: src/main/kotlin/net/hectus/neobb/modes/turn/person_game/block/PTCake.kt
package net.hectus.neobb.modes.turn.person_game.block

import com.marcpg.libpg.storing.Cord
import net.hectus.neobb.buff.Buff
import net.hectus.neobb.buff.Luck
import net.hectus.neobb.modes.turn.default_game.block.BlockTurn
import net.hectus.neobb.modes.turn.person_game.categorization.BuffCategory
import net.hectus.neobb.player.NeoPlayer
import org.bukkit.block.Block

class PTCake(data: Block?, cord: Cord?, player: NeoPlayer?) : BlockTurn(data, cord, player), BuffCategory {
    override fun buffs(): List<Buff<*>> {
        return listOf(Luck(20))
    }
}
